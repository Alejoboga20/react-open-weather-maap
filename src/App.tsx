import { Header } from './components/common/Header';
import { CurrentWeatherDisplay } from './components/WeatherDisplay/CurrentWeatherDisplay';
import { useApp } from './hooks/useApp';
import Loader from 'react-loader-spinner';
import { Label } from './components/common/Label';
import { PlaceSelector } from './components/PlaceSelector/PlaceSelector';
import { WeatherDisplay } from './components/WeatherDisplay/WeatherDisplay';
import './styles/App.css';

export const App = () => {
  const {
    isLoading,
    temp,
    inputButtons,
    handleOnClick,
    dailyTemp,
    blockButtons,
  } = useApp();

  return (
    <div className='app'>
      <Header title='React Weather App' subtitle='Code Challenge' />

      <div className='currentWeather_container'>
        {!isLoading ? (
          <CurrentWeatherDisplay temperature={temp} />
        ) : (
          <Loader type='TailSpin' color='#F38BA0' />
        )}
      </div>

      <div className='placeSelector_container'>
        <Label labelContent='Select the city to checkout its weather' />
        <PlaceSelector
          inputButtons={inputButtons}
          onClick={handleOnClick}
          blockButtons={blockButtons}
        />
      </div>

      <div className='weatherDisplayer_container'>
        {!blockButtons ? (
          <WeatherDisplay dailyTemp={dailyTemp} />
        ) : (
          <Loader type='TailSpin' color='#F38BA0' />
        )}
      </div>
    </div>
  );
};
