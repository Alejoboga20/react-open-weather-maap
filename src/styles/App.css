.app {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  background-color: #b5eaea;
  border: solid 2px black;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  margin: 3%;
  padding: 30px;
  text-align: center;
}
