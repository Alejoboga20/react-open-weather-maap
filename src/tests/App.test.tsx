import { mount } from 'enzyme';
import { App } from '../App';

describe('App Tests', () => {
  const wrapper = mount(<App />);

  test('should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should display 5 input Buttons', () => {
    expect(wrapper.find('button').length).toBe(5);
  });
});
