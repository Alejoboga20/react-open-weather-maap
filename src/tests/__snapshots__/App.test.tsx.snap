// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests should Match Snapshot 1`] = `
<div
  className="app"
>
  <div
    className="header_container"
  >
    <h1
      className="title"
    >
      React Weather App
    </h1>
    <h2
      className="subtitle"
    >
      Code Challenge
    </h2>
    <hr />
  </div>
  <div
    className="currentWeather_container"
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        height={80}
        viewBox="0 0 38 38"
        width={80}
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a"
            x1="8.042%"
            x2="65.682%"
            y1="0%"
            y2="23.865%"
          >
            <stop
              offset="0%"
              stopColor="#F38BA0"
              stopOpacity="0"
            />
            <stop
              offset="63.146%"
              stopColor="#F38BA0"
              stopOpacity=".631"
            />
            <stop
              offset="100%"
              stopColor="#F38BA0"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              id="Oval-2"
              stroke="#F38BA0"
              strokeWidth="2"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              fill="#fff"
              r={1}
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="placeSelector_container"
  >
    <div
      className="label_container"
    >
      <label
        className="main_content"
      >
        Select the city to checkout its weather
      </label>
    </div>
    <div
      className="radio_container"
    >
      <button
        className="inputButton"
        disabled={false}
        onClick={[Function]}
        value="BA"
      >
        Buenos Aires
      </button>
      <button
        className="inputButton"
        disabled={false}
        onClick={[Function]}
        value="ML"
      >
        Manila
      </button>
      <button
        className="inputButton"
        disabled={false}
        onClick={[Function]}
        value="MD"
      >
        Madrid
      </button>
      <button
        className="inputButton"
        disabled={false}
        onClick={[Function]}
        value="CN"
      >
        China
      </button>
      <button
        className="inputButton"
        disabled={false}
        onClick={[Function]}
        value="AU"
      >
        Auckland
      </button>
    </div>
  </div>
  <div
    className="weatherDisplayer_container"
  >
    <div
      className="weatherDisplay_container"
    />
  </div>
</div>
`;
